web/src/main/java/org/mskcc/cbio/oncokb/api/pvt/PrivateUtilsApiController.java

```diff
@@ -377,7 +378,18 @@ public ResponseEntity<VariantAnnotation> utilVariantAnnotationGet(
                 return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
             }
         }
-        if (StringUtils.isNullOrEmpty(hgvsg)) {
+        if (!StringUtils.isNullOrEmpty(hgvsg) || !StringUtils.isNullOrEmpty(genomicChange)) {
+            String genomicQuery = StringUtils.isNullOrEmpty(hgvsg) ? genomicChange : hgvsg;
+            GNVariantAnnotationType type = StringUtils.isNullOrEmpty(hgvsg) ? GNVariantAnnotationType.GENOMIC_LOCATION : GNVariantAnnotationType.HGVS_G;
+            Alteration alterationModel;
+            if (!this.cacheFetcher.genomicLocationShouldBeAnnotated(type, genomicQuery, matchedRG, this.cacheFetcher.getAllTranscriptGenes())) {
+                alterationModel = new Alteration();
+            } else {
+                alterationModel = this.cacheFetcher.getAlterationFromGenomeNexus(type, matchedRG, genomicQuery);
+            }
+            query = QueryUtils.getQueryFromAlteration(matchedRG, tumorType, alterationModel, hgvsg);
```

Comment:
Just a quick question, why do we save the HGVSg into the Query object and not the genomic change?

Reply:
It's a historical issue that I didn't include when rolling out the byGenomicChange annotation endpoint. I don't intend to fix it in this PR as the changes are related to a private endpoint. I usually don't change models that used in the published endpoints.